### PR TITLE
Fixes issue #721 - Support --http on Piranha Micro

### PR DESCRIPTION
**Pull request by Thihup**
Fixes issue #721

### CODE REVIEW

File: extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java [hunk 1]
```diff
@@ -185,8 +185,8 @@
             
             webApplicationServer.initialize();
             webApplicationServer.start();
-            
-            httpServer = new DefaultHttpServer(port, webApplicationServer, false);
```
Comment (manorrock) [inline extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: Why is false removed here? We need to be able to support running in SSL mode. I am not sure if we ported that from Piranha Server to Piranha Micro yet ...

File: extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java [hunk 2]
```diff
@@ -195,7 +195,20 @@
...
     }
-    
+
+    private HttpServer getHttpServer(int port, HttpServerProcessor serverProcessor) {
+        ServiceLoader<HttpServer> httpServers = ServiceLoader.load(HttpServer.class);
+        for (HttpServer server : httpServers) {
+            Class<? extends HttpServer> aClass = server.getClass();
+            try {
+                return aClass.getConstructor(int.class, HttpServerProcessor.class).newInstance(port, serverProcessor);
+            } catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException ignored) {
```
Comment (manorrock) [inline extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:205]: Please use a java.util Logger to show the reason why it failed ... use log level WARNING

File: http/impl/src/main/java/cloud/piranha/http/impl/DefaultHttpServer.java [hunk 3]
```diff
@@ -141,6 +141,21 @@ public DefaultHttpServer(int serverPort, HttpServerProcessor processor, boolean
...
     }
 
+
+    /**
+     * Constructor
+     *
+     * @param serverPort the server port.
+     * @param processor the HTTP server processor.
+     */
+    public DefaultHttpServer(int serverPort, HttpServerProcessor processor) {
```
Comment (manorrock) [inline http/impl/src/main/java/cloud/piranha/http/impl/DefaultHttpServer.java:151]: Please use the existing constructor that takes in a boolean to enable/disable SSL

File: micro/src/main/java/cloud/piranha/micro/MicroConfiguration.java [hunk 4]
```diff
@@ -126,9 +130,14 @@ public MicroConfiguration postConstruct() {
                 .filter(repo -> !repo.isEmpty())
                 .collect(toList());
 
-        mergedDependencies = concat(dependenciesFromExtensionsStream, directDependenciesStream)
-                .collect(toList());
-        
+
+        // A bit overkill
```
Comment (manorrock) [inline micro/src/main/java/cloud/piranha/micro/MicroConfiguration.java:134]: Why the comment?

File: extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java (hunk 1 above)
Comment (manorrock) [reply extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: Note if we did not completely port it yet then please just pass in false and we can address it at a later date

Comment (Thihup) [reply extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: As I'm creating the server through reflection, I'll have to try more than one constructor. If it is OK I can add right now

Comment (manorrock) [reply extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: OK lets rework the implementations to have only one constructor and then add getters/setters to the HttpServer API and adjust each implementation accordingly. Create everything needed in the default constructor and where that is not possible create it in the start method. I hope that makes sense?

Comment (Thihup) [reply extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: Looks good to me

Comment (Thihup) [reply extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java:189]: The default will be a no-arg constructor or a three arguments constructor (serverPort, processor, ssl) ?

**Pull request merged**